packages/repository/src/__tests__/unit/repositories/relation.helpers.unit.ts

```diff
@@ -3,113 +3,398 @@
-    }
-    expect(errorMessage).to.eql('scope is not supported');
+      const products = await findByForeignKeys(productRepo, 'categoryId', [2]);
+      expect(products).to.deepEqual([pencils]);
+      expect(products).to.not.containDeep(pens);
+    });
+
+    it('returns all instances that have any of multiple foreign key values', async () => {
+      const pens = await productRepo.create({name: 'pens', categoryId: 1});
+      const pencils = await productRepo.create({
+        name: 'pencils',
+        categoryId: 2,
+      });
+      const paper = await productRepo.create({name: 'paper', categoryId: 3});
+      const products = await findByForeignKeys(productRepo, 'categoryId', [
+        1,
+        3,
+      ]);
+      expect(products).to.deepEqual([pens, paper]);
+      expect(products).to.not.containDeep(pencils);
+    });
+
+    it('throws error if scope is passed in and is non-empty', async () => {
+      let errorMessage;
+      try {
+        await findByForeignKeys(productRepo, 'categoryId', [1], {
+          limit: 1,
+        });
+      } catch (error) {
+        errorMessage = error.message;
+      }
+      expect(errorMessage).to.eql('scope is not supported');
+    });
+
+    it('does not throw an error if scope is passed in and is undefined or empty', async () => {
+      let products = await findByForeignKeys(
+        productRepo,
+        'categoryId',
+        [1],
+        undefined,
+        {},
+      );
+      expect(products).to.be.empty();
+      products = await findByForeignKeys(productRepo, 'categoryId', 1, {}, {});
+      expect(products).to.be.empty();
+    });
   });
 
-  it('does not throw an error if scope is passed in and is undefined or empty', async () => {
-    let products = await findByForeignKeys(
-      productRepo,
-      'categoryId',
-      [1],
-      undefined,
-      {},
-    );
-    expect(products).to.be.empty();
-    products = await findByForeignKeys(productRepo, 'categoryId', 1, {}, {});
-    expect(products).to.be.empty();
+  describe('includeRelatedModels', () => {
+    beforeEach(async () => {
+      await productRepo.deleteAll();
+      await categoryRepo.deleteAll();
+    });
+
+    it("defines a repository's inclusionResolvers property", () => {
+      expect(categoryRepo.inclusionResolvers).to.not.be.undefined();
+      expect(productRepo.inclusionResolvers).to.not.be.undefined();
+    });
+
+    it('returns source model if no filter is passed in', async () => {
+      const category = await categoryRepo.create({name: 'category 1'});
+      await categoryRepo.create({name: 'category 2'});
+      const result = await includeRelatedModels(categoryRepo, [category]);
+      expect(result).to.eql([category]);
+    });
+
+    it('throws error if the target repository does not have the registered resolver', async () => {
+      const category = await categoryRepo.create({name: 'category 1'});
+      await productRepo.create({id: 1, name: 'product1', categoryId: 1});
```

Comment:
Do we need this product to be created? IIUC, this test is never accessing any products, therefore it's safe to remove this line from this test to make it simpler.

Reply:
Removed 👍